compiler/cpp/test/parser/ParserTest.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2021 Compose Generator Contributors. All rights reserved.
+
+#include <parser/Parser.h>
+#include <gtest/gtest.h>
+#include "../util/LanguageDescriptor.h"
+#include "../../lib/json/json.hpp"
+#include "../util/FileReader.h"
+
+using json = nlohmann::json;
+
+struct ParserParams {
+    const std::string fileName;
+};
+
+const ParserParams PARSER_TEST_PARAMETERS[] = {
+        {
+            "has-block"
+        },
+        {
+            "comp-block"
+        },
+        {
+            "contains-block"
+        }
```

Comment:
what about ifBlock, stmtList, index, key etc.?

Reply:
Those structures are already included in the test files, but I agree that the index token for example is missing in the test files. Will add the index to a test file.